docs/docsite/rst/dev_guide/developing_modules_network.rst

```diff
@@ -0,0 +1,51 @@
+.. _network_dev_guide:
+
+**************************
+Developing network modules
+**************************
+
+.. contents::
+   :local:
+
+.. _developing_modules_network:
+
+Network connection plugins
+==========================
+
+Each network connection plugin has a set of implementation plugins which provide a specific implementation of that connection for a particular set of devices.
```

Comment:
I don't like 'implementation plugins'. I appreciate that's the phrase, but it's confusing since 'implementation' is used later. Maybe just say "...has a set of plugins which provide a specification of the connection for a particular set of devices."